### DISCUSSION TITLE
How to use swr with Express

### DISCUSSION THREAD

**Discussion by hyphen1370** [vercel/swr#1513] [Q&A]
Hello, 

I am trying to build a react ssr on Azure function for Javascript with express. However, Azure function for Javascript doesn't support any ES6 or above. I have to use es5 syntax. Could I use swr in this environment? 

Here is what I tried: 
async function getChecklistDate() {
const useSWR = require('swr').useSWR;
const commonFunctions = require('../../components/commonFunctions/CommonFunctions.js');
const { request, gql } = require('graphql-request');

async function getChecklistDate() {
    const { data, error } = useSWR(
        `query backupRestoreChecklistGetCheckDateAllQuarter {
            backupRestoreChecklistGetCheckDateAllQuarter
        }`,
        query => request('/api/graphql', query),
    )
    console.log(data);
    console.log(error);
    return "data";
}

when I ran it, it gave me this error:
(node:23177) UnhandledPromiseRejectionWarning: TypeError: useSWR is not a function

Thanks

**Comment by shuding** [comment]
SWR is a React Hook, it seems that you're not running it inside a React Component?

**Reply by hyphen1370** [reply]
OK, I am trying to run in server side rendering inside express node, is swr also works with .mjs? 

Thanks

**Reply by shuding** [reply]
SWR has ES module export, so yes.
